File: samples/posix/bin/test-user.pl
#!/usr/bin/perl

if ($#ARGV == 0) {
    $name = $ARGV[0];
} else {
    $name = `/usr/bin/whoami`;
    chomp($name);
}

print "Calling getpwnam(\"$name\")...\n\n";

($name, $password, $uid, $gid, $quota, $comment, $gecos,
 $dir, $shell, $expire) = getpwnam($name);

print "Name     : $name\n";
print "Password : $password\n";
print "UID      : $uid\n";
print "GID      : $gid\n";
print "Quota    : $quota\n";
print "Comment  : $comment\n";
print "Gecos    : $gecos\n";
print "Home     : $dir\n";
print "Shell    : $shell\n";
print "Expire   : $expire\n";

File: samples/posix/src/test-user.c
#include <stdio.h>
#include <sys/types.h>
#include <pwd.h>

int main(int argc, char **argv) {

    if (argc != 2) {
        printf("Usage: test-user <name>\n");
        return -1;
    }

    char* name = argv[1];

    struct passwd* result = getpwnam(name);

    if (!result) {
        printf("User %s not found.\n", name);
        return -1;
    }

    printf("Name      : %s\n", result->pw_name);
    printf("Password  : %s\n", result->pw_passwd);
    printf("UID       : %d\n", result->pw_uid);
    printf("GID       : %d\n", result->pw_gid);
    printf("Gecos     : %s\n", result->pw_gecos);
    printf("Directory : %s\n", result->pw_dir);
    printf("Shell     : %s\n", result->pw_shell);
    printf("\n");

    return 0;
}
